service/history/timerQueueStandbyTaskExecutor.go

```diff
@@ -302,7 +302,7 @@ func (t *timerQueueStandbyTaskExecutor) executeActivityRetryTimerTask(
 			t.getCurrentTime,
 			t.config.StandbyTaskMissingEventsResendDelay(),
 			t.config.StandbyTaskMissingEventsDiscardDelay(),
-			t.pushActivity,
+			t.fetchHistoryFromRemote,
```

Comment:
Or only refresh? or even simply just no-op?

Reply:
Just arguing from load perspective - keep it no-op if possible. 
In case of replication delay, it seems to me that fetchHistoryFromRemote will amplify load.

Reply:
It depends on what is the common reason we need to fetch from remote? Is it replication delay or missing replication task? 
If it is replication delay, we probably should keep waiting, which is no-op. 
If it is missing task, then we should fetch, but we should only **fetch once**.

Reply:
Synced offline with @yiminc and @wxing1292 last week. We think resend is still needed here and for majority of the case after resending once, the task will be unblocked.

Using no-op is probably not a good idea as it will likely block the queue until the discard time and still send the task to matching.